tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -73,5 +75,79 @@ internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, s
 
             return string.Empty;
         }
+
+        /// <summary>
+        /// Uses a sql instruction to set a context for the current connection, bearing the span ID and trace ID.
+        /// This is meant to circumvent cache invalidation issues that occur when those values are injected in comment.
+        /// Currently only working for MSSQL (uses an instruction that is specific to it)
+        /// </summary>
+        /// <returns>True if the traceparent information was set</returns>
+        internal static bool PropagateDataViaContext(Tracer tracer, DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbConnection? connection, string serviceName, Scope scope, SqlTags tags)
+        {
+            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient || connection == null)
+            {
+                return false;
+            }
+
+            // we want the instrumentation span to be a sibling of the actual query span
+            var instrumentationParent = scope.Parent?.Span?.Context;
+            // copy the tags so that modifications on one span don't impact the other
+            var copyProcessor = new ITags.CopyProcessor<SqlTags>();
+            tags.EnumerateTags(ref copyProcessor);
+            using (var instrumentationScope = tracer.StartActiveInternal("set context_info", instrumentationParent, tags: copyProcessor.TagsCopy, serviceName: serviceName))
+            {
+                instrumentationScope.Span.Type = SpanTypes.Sql;
+                // this tag serves as "documentation" for users to realize this is something done by the instrumentation
+                instrumentationScope.Span.Tags.SetTag("dd.instrumentation", "true");
```

Comment:
What is `"dd.instrumentation"` for? If it's an internal tag, why doesn't it use the `_dd` prefix?

Reply:
It's not an internal tag, it's meant as some kind of documentation for the users to realize this is not a span representing work done by their app, but work done by DD, for instrumentation purposes.

We decided to have a span for this because it's probably the slowest instrumentation we've ever introduced so far

Reply:
Thanks